components/locale/src/fallback/algorithms.rs

```diff
@@ -187,12 +246,20 @@ mod tests {
             input: "en-u-sd-usca",
             requires_data: false,
             expected_language_chain: &["en-u-sd-usca", "en"],
+            expected_script_chain: &["en-u-sd-usca", "en", "und-u-sd-usca"],
             expected_region_chain: &["en-u-sd-usca", "en", "und-u-sd-usca"],
         },
         TestCase {
             input: "en-US-u-sd-usca",
             requires_data: false,
             expected_language_chain: &["en-US-u-sd-usca", "en-US", "en-u-sd-usca", "en"],
+            expected_script_chain: &[
+                "en-US-u-sd-usca",
+                "en-US",
+                "en-u-sd-usca",
+                "en",
+                "und-u-sd-usca",
```

Comment:
Issue: `und-Latn` should definitely be in the chain, and `und-u-sd-usca` probably not.

Reply:
This is `requires_data: false`. Without data we don't know that `en` is `Latn`. The same test exists with data below.